src/contexts/addresses.tsx

```diff
@@ -346,17 +338,26 @@ export const AddressesContextProvider: FC<{ overrideContextValue?: PartialDeep<A
     return () => {
       clearInterval(interval)
     }
-  }, [addressesState, currentNetwork, refreshAddressesState])
+  }, [
+    addressesState,
+    currentNetwork,
+    fetchAndStoreAddressesData,
+    addressesOfCurrentNetwork,
+    addressesWithPendingSentTxs
+  ])
 
   return (
     <AddressesContext.Provider
       value={merge(
         {
-          addressesState: getAddressesStateOfCurrentNetwork(),
+          addresses: addressesOfCurrentNetwork,
+          mainAddress: addressesOfCurrentNetwork.find((address) => address.settings.isMain),
+          getAddress,
+          setAddress,
```

Comment:
It would be great to make it clear when we're storing something in the local storage or not.
Also, we need to define if we want to specify Address**Data** all the time or not.

I would suggest to rename:
 `updateAddressSettings` into `updateAndStoreAddressSettings`
`refreshAddressesData` into `fetchAndStoreExistingAddresses`
`saveNewAddress` into `fetchAndStoreNewAddress`

Those are only propositions, not sure about all of them. WDYT?

Reply:
> I would suggest to rename:  
> `updateAddressSettings` into `updateAndStoreAddressSettings`  
> `refreshAddressesData` into `fetchAndStoreExistingAddresses`  
> `saveNewAddress` into `fetchAndStoreNewAddress`  


Hm, that's how I had initially named it. However, after a few code iterations, I thought it'd be nice to create a level of abstraction for the address context API. It made sense to me that the function that takes care of everything that needs to be done when generating a new adddress (store in local storage and fetch data from API) is called `save`.

Same goes for using `updateAddressSettings` that uses the word `update` instead of explicitly listing everything the function does in its name.

Same for `refreshAddressesData`.

I don't have a strong opinion on this, although I do prefer the names I chose. I fully understand your opinion that you prefer the function names to be as explicit as possible. My only argument is to have a (imo) cleaner context API.

```
          addresses,
          mainAddress,
          getAddress,
          setAddress,
          saveNewAddress,
          updateAddressSettings,
          refreshAddressesData,
          isLoadingData
```

VS

```
          addresses,
          mainAddress,
          getAddress,
          setAddress,
          fetchAndStoreNewAddress,
          updateAndStoreAddressSettings,
          fetchAndStoreExistingAddresses,
          isLoadingData
```

> It would be great to make it clear when we're storing something in the local storage or not.

Lastly, if we were to use the word `store` only for when we store things in local storage we should then agree to use the word `set` when storing things in the state. Based on this, we would then need to update the names to sth like this:

- `storeFetchAndSetNewAddress` (instead of `saveNewAddress` and `fetchAndStoreNewAddress`)
- `storeAndSetAddressSettings` (instead of `updateAddressSettings` and `updateAndStoreAddressSettings`)
- `fetchAndSetExistingAddresses` (instead of `refreshAddressesData` and `fetchAndStoreExistingAddresses`)

> Also, we need to define if we want to specify AddressData all the time or not.

"Fetching address data" sounds better to me than "fetch address", but again, no strong opinion.

---

I look forward to hearing your thoughts on all these.

Reply:
If you're fully comfortable with this level of abstraction, then I'm fine with it. My initial reaction was caused by the fact that the methods' names were not enough for me to clearly understand what these do: memory storage vs. local storage; remote fetch vs. offline... Just wanted to have your opinion on this, so thanks for that.

But it's true that the proposed naming is a mouthful! I wonder what Lee would say about this hehe ;) 

**TLDR: I trust your guts on this one.**